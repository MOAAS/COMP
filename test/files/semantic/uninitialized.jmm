import io.println(int) void;
import Quicksort;
import List;
import List(int);

class Simple extends Quicksort{
    int simpleVar;

    public static void main(String[] arg){
        int a;
        int b;
        int c;
        int d;
        int e;
        int z;
        boolean g;
        boolean h;
        Simple simp1;
        Simple simp2;
        List list;
        a = 1;
        simp1 = new Simple();
        if(simp1.getBool() && g){
            b = 2;
            d = 4;
        } else {
            b = 3;
            c = 3;
            if(!simp2.getBool())
                 d = 4;
            else d = 4;
        }

        while(true){
         g = false;
        }
        list = new List(e);
        e = a;
        e = b;
        e = c;
        e = d;
        h = g;
    }
    public int getInt(int i){
      int x;
      int a;
      a = x;
      x = a;
      x = i;
      return x;
    }

    public boolean getBool(){
      return true;
    }
 }