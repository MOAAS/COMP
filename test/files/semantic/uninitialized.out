Error: Variable "g" has not been initialized at line 23, column 31.
Error: Variable "simp2" has not been initialized at line 29, column 17.
Error: Variable "e" has not been initialized at line 37, column 25.
Warning: Variable "c" may not have been initialized at line 40, column 13.
Warning: Variable "g" may not have been initialized at line 42, column 13.
Error: Variable "x" has not been initialized at line 47, column 11.
Error: Variable "a" has not been initialized at line 48, column 11.
